esphome/components/uart/uart_esp8266.cpp

```diff
@@ -4,11 +4,17 @@
 #include "esphome/core/helpers.h"
 #include "esphome/core/application.h"
 #include "esphome/core/defines.h"
-#
+
+#ifdef USE_LOGGER
+#include "esphome/components/logger/logger.h"
+#endif
+
 namespace esphome {
 namespace uart {
 
 static const char *const TAG = "uart_esp8266";
+bool UARTComponent::serial0InUse = false;
```

Comment:
To suppress the clang warning:
```suggestion
bool UARTComponent::serial0InUse = false;  // NOLINT(cppcoreguidelines-avoid-non-const-global-variables)
```